src/win/util.c

```diff
@@ -1676,14 +1680,37 @@ int uv_os_gethostname(char* buffer, size_t* size) {
     return uv_translate_sys_error(WSAGetLastError());
 
   buf[sizeof(buf) - 1] = '\0'; /* Null terminate, just to be safe. */
-  len = strlen(buf);
+  utf16_str_buf_size = MultiByteToWideChar(CP_ACP, 0, buf, -1, NULL, 0);
```

Comment:
Another maintainer with more Windows experience should make the call on whether `CP_ACP` (system ANSI code page) is the right encoding. It might be, I don't know.

Reply:
Probably not documented (though it is perhaps supposed to be ASCII, as expected by DNS after punycode IDN conversion). But just use `GetHostNameW` above to entirely avoid the problem.

Reply:
@vtjnash GetHostNameW is
supported from win server2012 while gethostname supported from win sever 2003. Is it ok to change uv_os_hostname’s
minimum support version?

Reply:
Got it. Changed to use GetHostNameW instead. Plz review again